message_ix_models/tools/iea/weo.py

```diff
@@ -0,0 +1,672 @@
+        (df_cost_ratio.technology == "csp")
+        & (df_cost_ratio.r11_region.isin(["EEU", "FSU"])),
+        "cost_ratio",
+    ] = 0
+
+    # Assumption 2: For pulverized coal with CCS and IGCC with CCS in MEA,
+    # make cost ratio the same as in the FSU region
+    # TODO: this method to replace the values seems a little prone to errors,
+    # so probably best to change later
+    df_cost_ratio.loc[
+        (df_cost_ratio.cost_ratio.isnull()) & (df_cost_ratio.r11_region == "MEA"),
+        "cost_ratio",
+    ] = df_cost_ratio.loc[
+        (df_cost_ratio.r11_region == "FSU")
+        & (df_cost_ratio.technology.isin(["pulverized_coal_ccs", "igcc_ccs"]))
+    ].cost_ratio.values
+
+    # Assumption 3: For CSP in PAO, assume the same as NAM region (cost ratio == 1)
+    df_cost_ratio.loc[
+        (df_cost_ratio.technology == "csp") & (df_cost_ratio.r11_region.isin(["PAO"])),
+        "cost_ratio",
+    ] = 1
+
+    return df_cost_ratio
+
+
+"""
+Match each MESSAGEix technology with a WEO technology
+"""
+dict_weo_technologies = {
+    "coal_ppl": "steam_coal_subcritical",
+    "gas_ppl": "gas_turbine",
+    "gas_ct": "gas_turbine",
+    "gas_cc": "ccgt",
+    "bio_ppl": "bioenergy_large",
+    "coal_adv": "steam_coal_supercritical",
+    "igcc": "igcc",
+    "bio_istig": "igcc",
+    "coal_adv_ccs": "pulverized_coal_ccs",
+    "igcc_ccs": "igcc_ccs",
+    "gas_cc_ccs": "ccgt_ccs",
+    "bio_istig_ccs": "igcc_ccs",
+    "syn_liq": "igcc",
+    "meth_coal": "igcc",
+    "syn_liq_ccs": "igcc_ccs",
+    "meth_coal_ccs": "igcc_ccs",
+    "h2_coal": "igcc",
+    "h2_smr": "igcc",
+    "h2_bio": "igcc",
+    "h2_coal_ccs": "igcc_ccs",
+    "h2_smr_ccs": "igcc_ccs",
+    "h2_bio_ccs": "igcc_ccs",
+    "eth_bio": "igcc",
+    "eth_bio_ccs": "igcc_ccs",
+    "c_ppl_co2scr": "pulverized_coal_ccs",
+    "g_ppl_co2scr": "ccgt_ccs",
+    "bio_ppl_co2scr": "igcc_ccs",
+    "wind_ppl": "wind_onshore",
+    "wind_ppf": "wind_offshore",
+    "solar_th_ppl": "csp",
+    "solar_pv_I": "solarpv_buildings",
+    "solar_pv_RC": "solarpv_buildings",
+    "solar_pv_ppl": "solarpv_large",
+    "geo_ppl": "geothermal",
+    "hydro_lc": "hydropower_large",
+    "hydro_hc": "hydropower_small",
+    "meth_ng": "igcc",
+    "meth_ng_ccs": "igcc_ccs",
+    "coal_ppl_u": "steam_coal_subcritical",
+    "stor_ppl": "",
+    "h2_elec": "",
+    "liq_bio": "igcc",
+    "liq_bio_ccs": "igcc_ccs",
+    "coal_i": "ccgt_chp",
+    "foil_i": "ccgt_chp",
+    "loil_i": "ccgt_chp",
+    "gas_i": "ccgt_chp",
+    "biomass_i": "bioenergy_medium_chp",
+    "eth_i": "bioenergy_medium_chp",
+    "meth_i": "bioenergy_medium_chp",
+    "elec_i": "ccgt_chp",
+    "h2_i": "ccgt_chp",
+    "hp_el_i": "ccgt_chp",
+    "hp_gas_i": "ccgt_chp",
+    "solar_i": "solarpv_buildings",
+    "heat_i": "ccgt_chp",
+    "geo_hpl": "geothermal",
+    "nuc_lc": "nuclear",
+    "nuc_hc": "nuclear",
+    "csp_sm1_ppl": "csp",
+    "csp_sm3_ppl": "csp",
+}
+
+conversion_2017_to_2005_usd = 83.416 / 103.015
```

Comment:
We could update this in the iam-units package used throughout, but not urgent.
Please state the source for these values.